src/components/Navbar/Navbar.js

```diff
@@ -1,42 +1,61 @@
 import { Link } from 'gatsby';
 import { StaticImage } from 'gatsby-plugin-image';
-import React, { useContext } from 'react';
-import { useState } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 
 import SidebarDark from '../../assets/svgs/Sidebar_dark.svg';
 import SidebarLight from '../../assets/svgs/Sidebar_light.svg';
 import ThemeIconLight from '../../assets/svgs/ThemeIcon_dark.svg';
 import ThemeIconDark from '../../assets/svgs/ThemeIcon_light.svg';
 import { ThemeContext } from '../../context/ThemeContex';
+import { changeLanguage } from '../LanguageSwitching/LanguageSwitch';
 import { SidebarMenu } from '../SidebarMenu/SidebarMenu';
 import {
+  darkActiveButton,
+  darkInactiveButton,
   desktopMenu,
   desktopMenuDark,
   desktopMenuItem,
-  engLight,
   languageThemeWraper,
+  lightActiveButton,
+  lightInactiveButton,
   line,
   lineDark,
   logo,
   logoContainer,
   mainContainer,
   mainContainerDark,
   name,
-  plDark,
-  plLight,
   sidebar,
   sidebarContainer,
 } from './Navbar.module.scss';
 
 export function Navbar() {
   const [isSidebarMenuVisible, setIsSidebarMenuVisible] = useState(false);
+  const [language, setLanguage] = useState('');
+  const [buttonColorPL, setButtonColorPL] = useState('');
+  const [buttonColorENG, setButtonColorENG] = useState('');
   const { isDarkTheme, toggleTheme } = useContext(ThemeContext);
   const { theme } = useContext(ThemeContext);
+  const { t } = useTranslation();
 
   function toggleSidebarMenu() {
     setIsSidebarMenuVisible((prev) => !prev);
   }
 
+  const handleButtonChange = (lang) => {
+    setLanguage(lang);
+  };
+
+  useEffect(() => {
+    if (language === 'pl') {
+      setButtonColorPL(isDarkTheme ? darkActiveButton : lightActiveButton);
+      setButtonColorENG(isDarkTheme ? darkInactiveButton : lightInactiveButton);
```

Comment:
use theme() function provided in ThemeContext, look for usage in other components